network/wsNetwork.go

```diff
@@ -1125,6 +1136,19 @@ func (wn *WebsocketNetwork) ServeHTTP(response http.ResponseWriter, request *htt
 		challenge = wn.prioScheme.NewPrioChallenge()
 		responseHeader.Set(PriorityChallengeHeader, challenge)
 	}
+
+	idChalB64 := request.Header.Get(ProtocolConectionIdentityChallengeHeader)
+	idChal := IdentityChallengeFromB64(idChalB64)
+	peerPublicKey := idChal.Key
+	// if we already have a verified peer using this key, do not proceed
+	if wn.peersByID[peerPublicKey] != nil &&
+		wn.peersByID[peerPublicKey].identityVerified {
```

Comment:
see `checkIncomingConnectionVariables()` probably need to `response.WriteHeader(status code)` and write an error message